job-server/test/spark.jobserver/auth/WebApiWithAuthenticationSpec.scala

```diff
@@ -71,42 +72,62 @@ class WebApiWithAuthenticationSpec extends FunSpec with Matchers with BeforeAndA
     api.myRoutes
   }
 
-  private val routes = routesWithTimeout("1 s")
+  private val routesWithProxyUser = routesWithTimeout(true, "1 s")
+  private val routesWithoutProxyUser = routesWithTimeout(false, "1 s")
 
+  private val USER_NAME = "presidentskroob"
   // set to some valid user
-  private val authorization = new Authorization(new BasicHttpCredentials("presidentskroob", "12345"))
-  private val authorizationInvalidPassword = new Authorization(new BasicHttpCredentials("presidentskroob", "xxx"))
+  private val authorization = new Authorization(new BasicHttpCredentials(USER_NAME, "12345"))
+  private val authorizationInvalidPassword = new Authorization(new BasicHttpCredentials(USER_NAME, "xxx"))
   private val authorizationUnknownUser = new Authorization(new BasicHttpCredentials("whoami", "xxx"))
   private val dt = DateTime.parse("2013-05-29T00Z")
   private val jobInfo = JobInfo("foo-1", "context", JarInfo("demo", dt), "com.abc.meme", dt, None, None)
   private val ResultKey = "result"
 
+  private val addedContexts = new scala.collection.mutable.HashSet[String] with SynchronizedSet[String]
+
   class DummyActor extends Actor {
     import CommonMessages._
     import JobInfoActor._
     def receive = {
       case ListJars                       => sender ! Map()
       case GetJobStatus(id)               => sender ! jobInfo
       case GetJobResult(id)               => sender ! JobResult(id, id + "!!!")
-      case ContextSupervisor.ListContexts => sender ! Seq("context1", "context2")
+      case ContextSupervisor.ListContexts => sender ! addedContexts.toSeq
+      case ContextSupervisor.AddContext(name, _) =>
+        if (addedContexts.contains(name)) {
+          sender ! ContextSupervisor.ContextAlreadyExists
+        } else {
+          addedContexts.add(name)
+          sender ! ContextSupervisor.ContextInitialized
+        }
+      case ContextSupervisor.StopContext(name) =>
+        addedContexts.remove(name)
+        sender ! ContextSupervisor.ContextStopped
+      case ContextSupervisor.AddContextsFromConfig =>
+        sender ! "OK"
+      case m =>
+        //dev support when adding new test cases:
+        println("Unhandled message: " + m)
```

Comment:
you mean 'sys.error("Unhandled message: " + m)' ? 
This is a test class after all. I could also just remove that case.